tests/pytests/test_contains.py

```diff
@@ -0,0 +1,334 @@
+    assertInfoField(env, 'idx_sortable', 'attributes', res_info)
+
+    # nostem 1st
+    env.expect('ft.create', 'idx_nostem1', 'schema', 't', 'TEXT', 'WITHSUFFIXTRIE', 'NOSTEM').ok()
+    res_info = [['identifier', 't', 'attribute', 't', 'type', 'TEXT', 'WEIGHT', '1', 'NOSTEM', 'WITHSUFFIXTRIE']]
+    assertInfoField(env, 'idx_nostem1', 'attributes', res_info)
+
+    # nostem 2nd
+    env.expect('ft.create', 'idx_nostem2', 'schema', 't', 'TEXT', 'NOSTEM', 'WITHSUFFIXTRIE').ok()
+    assertInfoField(env, 'idx_nostem2', 'attributes', res_info)
+
+def testWITHSUFFIXTRIEParamTag(env):
+    conn = getConnectionByEnv(env)
+    env.expect('ft.create', 'idx', 'schema', 't', 'TAG', 'SORTABLE', 'WITHSUFFIXTRIE').error()
+    env.expect('ft.create', 'idx', 'schema', 't', 'TAG', 'SORTABLE', 'CASESENSITIVE').error() # sortable must be last
+
+    # without sortable
+    env.expect('ft.create', 'idx', 'schema', 't', 'TAG', 'WITHSUFFIXTRIE').ok()
+    res_info = [['identifier', 't', 'attribute', 't', 'type', 'TAG', 'SEPARATOR', ',', 'WITHSUFFIXTRIE']]
+    assertInfoField(env, 'idx', 'attributes', res_info)
+
+    # with sortable
+    env.expect('ft.create', 'idx_sortable', 'schema', 't', 'TAG', 'WITHSUFFIXTRIE', 'SORTABLE').ok()
+    res_info = [['identifier', 't', 'attribute', 't', 'type', 'TAG', 'SEPARATOR', ',', 'SORTABLE', 'WITHSUFFIXTRIE']]
+    assertInfoField(env, 'idx_sortable', 'attributes', res_info)
+
+    # with casesensitive
+    env.expect('ft.create', 'idx_casesensitive', 'schema', 't', 'TAG', 'WITHSUFFIXTRIE', 'CASESENSITIVE', 'SORTABLE').ok()
+    res_info = [['identifier', 't', 'attribute', 't', 'type', 'TAG', 'SEPARATOR', ',', 'CASESENSITIVE', 'SORTABLE', 'WITHSUFFIXTRIE']]
+    assertInfoField(env, 'idx_casesensitive', 'attributes', res_info)
+
+def testBasicContains(env):
+    conn = getConnectionByEnv(env)
+
+    env.expect('ft.create', 'idx', 'schema', 'title', 'text', 'body', 'text').ok()
+    conn.execute_command('HSET', 'doc1', 'title', 'hello world', 'body', 'this is a test')
+
+    # prefix
+    res = env.execute_command('ft.search', 'idx', 'worl*')
+    env.assertEqual(res[0:2], [1, 'doc1'])
+    env.assertEqual(set(res[2]), set(['title', 'hello world', 'body', 'this is a test']))
+
+    # suffix
+    res = env.execute_command('ft.search', 'idx', '*orld')
+    env.assertEqual(res[0:2], [1, 'doc1'])
+    env.assertEqual(set(res[2]), set(['title', 'hello world', 'body', 'this is a test']))
+    env.expect('ft.search', 'idx', '*orl').equal([0])
+
+    # contains
+    res = env.execute_command('ft.search', 'idx', '*orl*')
+    env.assertEqual(res[0:2], [1, 'doc1'])
+    env.assertEqual(set(res[2]), set(['title', 'hello world', 'body', 'this is a test']))
+
+def testSanity(env):
+    env.skipOnCluster()
+    env.expect('ft.config', 'set', 'MINPREFIX', 1).ok()
+    env.expect('ft.config', 'set', 'TIMEOUT', 100000).ok()
+    env.expect('ft.config', 'set', 'MAXEXPANSIONS', 10000000).equal('OK')
```

Comment:
Let add a test that check timeout with both OnTImeout policies...

Reply:
Added tests.With contains, we always fail b/c we timeout while creating the iterator and before we start to read results from them.
Creating a test